factory/src/main/java/com/agileactors/factory/v2/meansoftransportation/Boat.java

```diff
@@ -0,0 +1,33 @@
+package com.agileactors.factory.v2.meansoftransportation;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+public class Boat {
+
+  private static final Double MAX_CARGO_LOAD = 13000.00;
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+  private String name;
+
+  public void checkCargo(String cargo) {
+    logger.info("Checking what is needed to load cargo of type {} into the boat", cargo);
+  }
+
+  public void loadCargo(Double quantity) {
+    if (quantity.compareTo(MAX_CARGO_LOAD) > 0) {
+      logger.info("Loaded cargo into several boats to be transported.");
```

Comment:
How about adding the quantity information at the logging? It will help in a concurrent env